low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,388 @@
+            req_start_time = int(time.mktime(time.strptime(
+                req_register.get(["%s>time" % req_id])[0],
+                '%Y-%m-%d %H:%M:%S')))
+            current_time = int(time.time())
+            is_req_expired = (current_time - req_start_time) > expiry_sec
+            if is_req_expired or (last_reboot > req_start_time and
+                                  status is BundleRequestHandler.IN_PROGRESS):
+                # Set request state as failed
+                BundleRequestHandler._update_request_status(
+                    req_id, "Failed - request is expired.")
+            status = req_register.get(["%s>status" % req_id])[0]
+
+        return status
+
+    def _update_request_status(req_id, status):
+        """Updates processed request information."""
+        req_info = req_register.get(["%s" % req_id])[0]
+        req_info.update({
+            "status": status,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        })
+        req_register.set(["%s" % req_id], [req_info])
+
+    @staticmethod
+    def _add_bundle_request(req_id, url):
+        """Updates new request information."""
+        req_info = {
+            "status": BundleRequestHandler.IN_PROGRESS,
+            "url": url,
+            "time": datetime.strftime(
+                datetime.now(), '%Y-%m-%d %H:%M:%S')
+        }
+        req_register.set(["%s" % req_id], [req_info])
+
+
+class SupportBundle(object):
+    """SSPL support bundle class"""
+
+    def __init__(self):
+
+        self.SYS_INFORMATION = "SYSTEM_INFORMATION"
+        self.IEM_SENSOR = "IEMSENSOR"
+        self._tmp_dir = '/tmp/cortx'
+        self._default_path = '%s/support_bundle/' % self._tmp_dir
+        self.sspl_log_default = f"/var/log/{sspl_const.PRODUCT_FAMILY}/sspl"
+        self.iem_log_default = f"/var/log/{sspl_const.PRODUCT_FAMILY}/iem"
+        self.sspl_conf_dir = f"{sspl_const.SSPL_SB_TMP}/sspl_conf/"
+        self.ipmi_sel_data = "{0}/ipmi_sel_data_{1}.txt".format(
+            sspl_const.SSPL_SB_TMP, str(int(time.time())))
+        self.boot_drvs_dta = "{0}/Server_OS_boot_drives_SMART_data_{1}".format(
+            sspl_const.SSPL_SB_TMP, str(int(time.time())))
+        self.enclosure_log = f"{sspl_const.SSPL_SB_TMP}/enclosure_logs.zip"
+        self.enclosure_wwn = "NA"
+        self.sspl_log_dir = Conf.get(SSPL_CONF, "%s>sspl_log_file_path" %
+                                    (self.SYS_INFORMATION)).replace("/sspl.log", "")
+        self.iem_log_dir = Conf.get(SSPL_CONF, "%s>log_file_path" %
+                                    (self.IEM_SENSOR)).replace("/iem_messages", "")
+
+    def generate(self, args):
+        """Generate support bundle."""
+        logger.info("Starting SSPL bundle generation")
+        self.prepare_bundle(args.bundle_id, args.target_path, args.noencl)
+        self.__clear_tmp_files()
+        status = SupportBundle().get_status(args.bundle_id)
+        print(status)
```

Comment:
I assume print statement is for testing purpose?
Please remove it in the final PR.

Reply:
Yes, I will remove in final code before merge.

Reply:
plz remove prints now

Reply:
Done